Firebase/CoreDiagnostics/ProtoSupport/Protos/firebasecore.options

```diff
@@ -0,0 +1,35 @@
+# Copyright 2019 Google
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#      http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# 
+
+# Options for firebasecore.proto
+
+logs.proto.mobilesdk.ios.ICoreConfiguration.version_name type:FT_POINTER
```

Comment:
I found that I actually preferred the options file because it's explicit rather than implicit. I think if we unify our nanopb generation infrastructure (something on my TODO list), I'll probably follow Firestore's example and lose the options files. I think I stuck with the options files because I was planning on limiting the length of some string fields and that conflicted with some of the stuff we did in our generator.

Reply:
Yeah, if you take our approach, it's an all-or-nothing affair. If you want to have limits on some of your string fields, then you'll have to be more subtle that what we're doing. (But it might be worth checking if that extra complexity is actually buying you all that much.)